### PR TITLE
Add `aws iam group policy attachment`

### PR DESCRIPTION
**Pull request by eliecharra**
| Q                 | A
| ----------------- | ---
| 🐛 Bug fix?       | no
| 🚀 New feature?   | yes
| ⚠ Deprecations?   | no
| ❌ BC Break       | no
| 🔗 Related issues | #1456 
| ❓ Documentation  | https://github.com/snyk/driftctl-docs/pull/241

## Description

⚠️ Please review and test locally with attention

### CODE REVIEW

File: pkg/resource/aws/testdata/acc/aws_iam_group_policy/.driftignore [hunk 1]
```diff
@@ -1,2 +1,2 @@
 *
-!aws_iam_group_policy
+!aws_iam_group
```
Comment (wbeuil) [inline pkg/resource/aws/testdata/acc/aws_iam_group_policy/.driftignore:2]: I don't understand why changed the .driftignore of the `aws_iam_group_policy` test, can you explain?

File: pkg/resource/aws/testdata/acc/aws_iam_group_policy/terraform.tf [hunk 2]
```diff
@@ -29,4 +29,3 @@ resource "aws_iam_group_policy" "my_developer_policy" {
         ]
     })
 }
-
```
Comment (wbeuil) [inline pkg/resource/aws/testdata/acc/aws_iam_group_policy/terraform.tf:32]: Useless noise here

**Review by wbeuil — CHANGES_REQUESTED** [review]
Small comments and I think you may need to change back the terraform lock file of the group policy test and rerun the acc test of the group policy attachment one.

**Comment by eliecharra** [comment]
@wbeuil I was rebased on an old version of the PR that added `aws_iam_group` #1477 that's why the diff was broken.

I removed the commit and rebased that one over main and thus that should be better now.

**Review by wbeuil — APPROVED** [review]
LGTM

**Pull request merged**